File: backend/Squirrel.Shared/DTO/DatabaseItem/DatabaseItemContentCompare.cs
using DiffPlex.DiffBuilder;
using Squirrel.Shared.DTO.Text;
using Squirrel.Shared.Enums;

namespace Squirrel.Shared.DTO.DatabaseItem;

public sealed class DatabaseItemContentCompare
{
    public string SchemaName { get; set; } = null!;
    public string ItemName { get; set; } = null!;
    
    public DatabaseItemType ItemType { get; set; }
    
    public SideBySideDiffResultDto? SideBySideDiff { get; set; }
    public InLineDiffResultDto? InLineDiff { get; set; }
}

File: frontend/src/app/models/database-items/database-item-content-compare.ts
import { InLineDiffResultDto } from '../text-pair/inline-diff-result-dto';
import { TextPairDifferenceDto } from '../text-pair/text-pair-difference-dto';

import { DatabaseItemType } from './database-item-type';

export interface DatabaseItemContentCompare {
    schemaName: string;
    itemName: string;
    itemType: DatabaseItemType;
    sideBySideDiff: TextPairDifferenceDto;
    inLineDiff: InLineDiffResultDto;
}
